analysis/src/CreateInterface.ml

```diff
@@ -213,6 +236,60 @@ let printSignature ~extractor ~signature =
       Buffer.add_string buf (indent ^ "@react.component\n");
       Buffer.add_string buf (indent ^ newItemStr ^ "\n");
       processSignature ~indent rest
+    | Sig_type
+        ( propsId,
+          {
+            type_params;
+            type_kind = Type_record (labelDecls, Record_optional_labels optLbls);
+          },
+          _ )
+      :: Sig_value (makeId (* make *), makeValueDesc)
+      :: rest
+      when Ident.name propsId = "props"
+           && getComponentTypeV4 makeValueDesc.val_type <> None
+           && optLbls |> List.mem "key" ->
+      (* PPX V4 component declaration:
+         type props = {..., key?: _}
+         let v = ...
```

Comment:
This is a heuristic.
Can't be super precise when working with the results of running a PPX.